### PR TITLE
Format code coverage results

### PR DESCRIPTION
**Pull request by RussKie**
* Display when coverage increases
* Better present failures

image

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4143)

### CODE REVIEW

**Comment by RussKie** [comment]
image

File: eng/scripts/ValidateProjectCoverage.ps1 [hunk 1]
```diff
@@ -83,27 +84,36 @@ $CoberturaReport.coverage.packages.package | ForEach-Object {
...
-                        "Project"=$Name;"Coverage Type"="Branch";
-                        "Actual"=$BranchCoverage;"Expected"=$MinCodeCoverage
+                        "Project"=$Name;"Coverage Type"="Branch";"Expected"=$MinCodeCoverage;"Actual"="$esc[1m$esc[0;31m$($BranchCoverage)$esc[0m"
+                    }
+                )
+            )
+        }
+
+        # Detect the over-coverage
+        [double]$lowerstReported = if ($LineCoverage -lt $BranchCoverage) { $LineCoverage } else { $BranchCoverage };
```
Comment (danmoseley) [inline eng/scripts/ValidateProjectCoverage.ps1:111]:
```suggestion
        [double]$lowestReported = if ($LineCoverage -lt $BranchCoverage) { $LineCoverage } else { $BranchCoverage };
```

File: eng/scripts/ValidateProjectCoverage.ps1 [hunk 2]
```diff
@@ -83,27 +84,36 @@ $CoberturaReport.coverage.packages.package | ForEach-Object {
...
+        if ($MinCodeCoverage -lt $lowerstReported) {
```
Comment (danmoseley) [inline eng/scripts/ValidateProjectCoverage.ps1:112]:
```suggestion
        if ($MinCodeCoverage -lt $lowestReported) {
```

Comment (RussKie) [reply eng/scripts/ValidateProjectCoverage.ps1:112]: Thank you!

**Pull request merged**